### ISSUE TITLE
INetworkQueryExtension::onAddRule needs to be called every onTick

### ISSUE THREAD

**Issue by japajoe** [openmultiplayer/open.mp#647]
As discussed on Discord, there appears to be a bug with the INetworkQueryExtension::addRule method. For example take this code

```cpp
for (INetwork *network : core->getNetworks())
{
    INetworkQueryExtension* query = queryExtension<INetworkQueryExtension>(network);

    if (query)
    {
        query->addRule("worldtime", "Monday 12:30");
    }
}   
```

The expected behavior is that you see the text **Monday 12:30** in the SA-MP client, but unless you call this method every onTick, it will not display/change at all. Initially I was calling the method in 1 seconds intervals but I noticed I couldn't see this value in the client. The solution was as I described earlier, to call it every onTick.

Edit: While trying to remove a rule with INetworkQueryExtension::removeRule I've noticed the same behavior, not sure whether this is intended or not.

**Comment by ksenonadv** [comment]
addRule works properly.
The server is overwriting worldtime rule each time there's an update triggered by ICore::updateNetworks()

**Comment by japajoe** [comment]
According to one of the devs on Discord it shouldn't work like this and he told me to open an issue. I don't have a dog in this race however but if this is the desired behavior then it should be documented.

**Comment by Y-Less** [comment]
PR #661

**Comment by AmyrAhmady** [comment]
Since Y_Less's PR is now closed, Fixed in PR: #683

**Comment by japajoe** [comment]
Beautiful thanks.
